internal/gatewayapi/filters.go

```diff
@@ -617,24 +619,41 @@ func (t *Translator) processResponseHeaderModifierFilter(
 	}
 }
 
-func (t *Translator) processExtensionRefHTTPFilter(filter v1beta1.HTTPRouteFilter, filterContext *HTTPFiltersContext, resources *Resources) {
+func (t *Translator) processExtensionRefHTTPFilter(ruleIdx int,
+	filter v1beta1.HTTPRouteFilter,
+	filterIdx int,
+	filterContext *HTTPFiltersContext,
```

Comment:
can be https://github.com/envoyproxy/gateway/blob/488c44bbfdfcce3e8e1496e03016d9a36d7efd1b/internal/gatewayapi/filters.go#L36 be enhanced to also store `filterIdx` ? similar to what is done here 
https://github.com/envoyproxy/gateway/blob/488c44bbfdfcce3e8e1496e03016d9a36d7efd1b/internal/gatewayapi/contexts.go#L50